### PR TITLE
Re-initialize type service

### PR DESCRIPTION
**Pull request by haritha-mohan**
test suite needs some work on organization..
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/xcsync/pull/129)

### CODE REVIEW

File: test/xcsync.e2e.tests/usecases/GenerateThenSyncWithChangesTests.cs [hunk 1]
```diff
@@ -59,6 +65,74 @@ public async Task GenerateThenSync_WithChanges_GeneratesChangesAsync (string pro
 			Assert.Fail ($"[{projectType},{tfm}] : Git diff-index failed, there are no changes in the source files.\n{commandOutput.Output}");
 	}
 
+	[Theory]
+	[MemberData (nameof (AddXcodeDotnetChanges))]
+	[Trait ("Category", "IntegrationTest")]
+	public async Task GenerateSync_MultipleFlowTest (string projectType, string tfm, Func<ITestOutputHelper, string, string, string, Task> xcodeChanges, Func<ITestOutputHelper, string, string, string, string, Task> dotnetChanges)
```
Comment (mcumming) [inline test/xcsync.e2e.tests/usecases/GenerateThenSyncWithChangesTests.cs:71]:
```suggestion
	public async Task GenerateAndSync_WithXcodeAndDotnetChanges_GeneratesChangesAsync (string projectType, string tfm, Func<ITestOutputHelper, string, string, string, Task> xcodeChanges, Func<ITestOutputHelper, string, string, string, string, Task> dotnetChanges)
```

File: test/xcsync.e2e.tests/usecases/GenerateThenSyncWithChangesTests.cs [hunk 2]
```diff
@@ -59,6 +65,74 @@ public async Task GenerateThenSync_WithChanges_GeneratesChangesAsync (string pro
...
+		await Xcsync (TestOutput, "generate", "--project", csproj, "--target", xcodeDir, "-tfm", tfm).ConfigureAwait (false);
+		await Git (TestOutput, "-C", tmpDir, "add", ".").ConfigureAwait (false);
+		await Git (TestOutput, "-C", tmpDir, "commit", "-m", "Xcode Project Generation").ConfigureAwait (false);
+
+		await xcodeChanges (TestOutput, xcodeDir, projectType, tfm).ConfigureAwait (false);
+		await Git (TestOutput, "-C", tmpDir, "add", ".").ConfigureAwait (false);
+		await Git (TestOutput, "-C", tmpDir, "commit", "-m", "Xcode Project Changes").ConfigureAwait (false);
+
+		await Xcsync (TestOutput, "sync", "--project", csproj, "--target", xcodeDir, "-tfm", tfm).ConfigureAwait (false);
+
```
Comment (mcumming) [inline test/xcsync.e2e.tests/usecases/GenerateThenSyncWithChangesTests.cs:101]: You should check that there are changes here.

File: test/xcsync.e2e.tests/usecases/GenerateThenSyncWithChangesTests.cs [hunk 3]
```diff
@@ -59,6 +65,74 @@ public async Task GenerateThenSync_WithChanges_GeneratesChangesAsync (string pro
...
+		await Git (TestOutput, "init", tmpDir).ConfigureAwait (false);
+		await DotnetNew (TestOutput, projectType, tmpDir, string.Empty).ConfigureAwait (false);
+		await DotnetNew (TestOutput, "gitignore", tmpDir, string.Empty).ConfigureAwait (false);
+		await Git (TestOutput, "-C", tmpDir, "add", ".").ConfigureAwait (false);
+		await Git (TestOutput, "-C", tmpDir, "commit", "-m", "Initial commit").ConfigureAwait (false);
+
+		// Act
+		await Xcsync (TestOutput, "generate", "--project", csproj, "--target", xcodeDir, "-tfm", tfm).ConfigureAwait (false);
+		await Git (TestOutput, "-C", tmpDir, "add", ".").ConfigureAwait (false);
```
Comment (mcumming) [inline test/xcsync.e2e.tests/usecases/GenerateThenSyncWithChangesTests.cs:93]: You should check that there are changes here.

**Review by mcumming — CHANGES_REQUESTED** [review]

File: src/xcsync/Workers/ChangeWorker.cs [hunk 4]
```diff
@@ -21,7 +21,7 @@ public override async Task ConsumeAsync (ChangeMessage message, CancellationToke
 		if (message.Direction == SyncDirection.ToXcode)
 			XcodeCommand<GenerateCommand>.RecreateDirectory (FileSystem, TargetDir);
 		Logger.Debug (Strings.Watch.Syncing);
-		await new SyncContext (FileSystem, TypeService, message.Direction, ProjectPath, TargetDir, Framework, Logger).SyncAsync (cancellationToken);
+		await new SyncContext (FileSystem, new TypeService (Logger!), message.Direction, ProjectPath, TargetDir, Framework, Logger).SyncAsync (cancellationToken);
```
Comment (stephen-hawley) [inline src/xcsync/Workers/ChangeWorker.cs:24]: Whenever I see a `!` assertion, I ask if this is avoidable.
In this case, I see the `Logger` variable being used several time before the assertion. This tells me that if `Logger` is null, this code will crash with a NRE before getting to the `!`.
I did a quick check on the source to serilog and it looks like there's no protection on the method. For example, if you implemented `Debug` like this:
```
public static void MyDebug(this ILogger? logger, string message)
{
    if (logger is null) return; // or exception?
    logger.Debug(message);
}
```
But it's not - it's a straight method.
What I would consider is:
1. put `#nullable enable` on this file and refactor appropriately.
2. if it makes sense for the `ILogger` to be null, then handle it appropriately before the `Debug` calls, the analyzer should be able to see that and not issue a warning so you shouldn't need the `!`; if not then the `!` isn't needed.

**Review by stephen-hawley — CHANGES_REQUESTED** [review]

**Review by mcumming — APPROVED** [review]

**Review by stephen-hawley — APPROVED** [review]

**Pull request merged**